_articles/scripting/Typescript/typescript-introduction.md

```diff
@@ -0,0 +1,164 @@
+
+![Click on path](../../../static/images/typescript-tutorial/clickAddressBar.png)
+
+* While the path is highlighted, delete the text and instead type "cmd" and press Enter. This will open the Command Shell with the folder path shown right before your input cursor.
+
+* Type `npm install` and press enter. You'll see a loading bar, which should take up to a minute to complete. If it was successful, you should see that your `game` and `content` folders now have a "shortcut" icon next to them. They're now symlinked to your Dota 2 folder. The `typescript_tutorial` folder now exists for both `game/dota_addons/` and `content/dota_addons` in your Dota 2 folder.
+
+:::note
+Symlinked folders are copies of each other, where any action done on one is also applied on the other folder. Therefore, you can simply work on your conveniently placed folder (`Desktop/Dota 2 custom games/typescript-example` in my example), and it will automatically be applied on the Steam Dota 2 folder as well.
+:::
+
+### Typescript Addon Structure
+Most of the structure for Typescript is identical to a standard Dota addon, such as having `scripts/vscripts/` folder. However, there are a few additional files responsible for making Typescript identify and work with Dota 2 API, and in the custom game in general.
+
+:::note
+The structure inside the `/game/` and `/content/` folders can be changed and designed as you wish; however, you will have to make sure all imports still point to the correct path. You should not change folders in the root folder unless you know what you're doing.
+:::
+
+Most of those files can be left alone, but it's good to know what they do. Most of those files are located at `/game/vscripts/` folder:
+* `tsconfig.json`: Configures how Typescript works in the lua portion of the project.
+
+:::note
+The same file exists in the `/content/panorama/scripts/custom_game` folder for Typescript configuration rules for panorama.
+::::
+
+* `vscripts/lib/dota_ts_adapter.ts`: Responsible for registering various common classes, such as abilities, items, and modifiers.
+* `vscripts/lib/tstl-utils.ts`: Responsible for the typescript-to-lua translation.
+* `vscripts/lib/timers.lua` and `vscripts/lib/timers.d.ts`: The common Timers library is already included in default in its lua form, with timers.d.ts including an interface to allow using the Timers library in Typescript.
+
+:::note
+An additional important folder is located at `/node_modules/dota-lua-types/types/`. This folder holds files that are responsible for all the API that exists for Dota 2 custom games. You can modify this file if you want. However, note that when updating the project (explained later), those files would be updated as well, effectively removing your changes. In addition, those are ignored in your Github, so others will not see your changes.
+:::
+
+### Updating Your Addon
```

Comment:
I would suggest moving this down because this is only sometimes relevant.

Reply:
Will push it to the end of the article, just before the dedicated help section.